modules/images/webp-uploads/load.php

```diff
@@ -4,41 +4,414 @@
+ * @param array|null $backup_sizes  An array with the current backup sizes, if not provided it would be queried against the meta data.
+ *
+ * @return array An array with the updated structure for the metadata before is stored in the database.
+ */
+function webp_uploads_create_images_with_additional_mime_types( array $metadata, $attachment_id, $backup_sizes = null ) {
+	// This should take place only on the JPEG image.
+	$valid_mime_types = webp_uploads_valid_image_mime_types();
+
+	// Not a supported mime type to create the sources property.
+	if ( ! array_key_exists( get_post_mime_type( $attachment_id ), $valid_mime_types ) ) {
+		return $metadata;
+	}
+
+	// If no size is present there's no need to generate any additional image as a backup.
+	if ( empty( $metadata['sizes'] ) ) {
+		return $metadata;
+	}
+
+	// All subsizes are created out of the `file` property and not the original image.
+	$file = get_attached_file( $attachment_id, true );
+
+	// File does not exist.
+	if ( ! file_exists( $file ) ) {
+		return $metadata;
+	}
+
+	if ( ! is_array( $backup_sizes ) ) {
+		$backup_sizes = get_post_meta( $attachment_id, '_wp_attachment_backup_sizes', true );
+		$backup_sizes = is_array( $backup_sizes ) ? $backup_sizes : array();
+	}
+
+	$dirname = pathinfo( $file, PATHINFO_DIRNAME );
+	// Find the hash of the file name to use it as a identifier if present.
+	preg_match( '/-e([\d]{13})\./', $file, $matches );
+
+	// $matches would have at least 2 values if the regex above matches.
+	if ( count( $matches ) >= 2 ) {
+		$hash = $matches[1];
+	} else {
+		$hash = time() + mt_rand( 100, 999 );
+	}
+
+	foreach ( webp_uploads_get_image_sizes() as $size => $properties ) {
+		// Generate backups only for the missing mime types.
+		$formats = webp_uploads_get_remaining_image_mimes( $metadata, $size );
+
+		foreach ( $formats as $mime => $extension ) {
+			$key = $extension . '-' . $size;
+
+			// The file already exists as part of the backup sizes in the same key.
+			if ( array_key_exists( $key, $backup_sizes ) ) {
+				// This case would remove the file if is an edited version.
+				if ( defined( 'IMAGE_EDIT_OVERWRITE' ) && IMAGE_EDIT_OVERWRITE ) {
+					$file_name     = empty( $backup_sizes[ $key ]['file'] ) ? '' : $backup_sizes[ $key ]['file'];
+					$file_location = path_join( $dirname, $file_name );
+					// Test to make sure this is an edited version, if so we can remove it safely.
+					preg_match( '/-e[\d]{13}-/', $file_name, $matches );
+					if ( ! empty( $matches ) && file_exists( $file_location ) ) {
+						wp_delete_file( $file_location );
+					}
+				} else {
+					$backup_sizes[ "{$key}-{$hash}" ] = $backup_sizes[ $key ];
+				}
+				// Clear the key, so the subsequent section can add the new image there.
+				unset( $backup_sizes[ $key ] );
+			}
+
+			// Editor needs to be recreated every time as there is not flush() or clear() function that can be used after we created an image.
```

Comment:
Did you try`$editor->__destruct` (or even just $editor = null which should trigger `__destruct`)?  For both Imagick and GD this should clear the image.

Reply:
No, let me try that thanks for the suggestion.

Reply:
Actually calling `__destruct` seems like the same as creating a new variable altogether as essentially this is called by the engine itself so, for now, will keep it like that. 

One thing that I've noted out of the class `WP_Image_Editor` is the fact that has a property `get_size` but not a `set_size` which would be the one we can use here to reset the size to the appropriate dimensions. Any thoughts on that one ?